### ISSUE TITLE
Cannot use HttpServer with UDS

### ISSUE THREAD

**Issue by andig** [reactphp/http#231]
PHP-PM uses UDS for handling master requests on slave. This fails in the `RequestHeaderParser`:

	parser->error Symfony\Component\Debug\Exception\ContextErrorException: Notice: Undefined index: port in /Users/andig/Documents/htdocs/vz/vendor/react/http/src/RequestHeaderParser.php:107
	Stack trace:
	#0 /Users/andig/Documents/htdocs/vz/vendor/react/http/src/RequestHeaderParser.php(59): React\Http\RequestHeaderParser->parseRequest('GET /data.json ...')
	#1 /Users/andig/Documents/htdocs/vz/vendor/react/http/src/RequestHeaderParser.php(49): React\Http\RequestHeaderParser->parseAndEmitRequest(132)
	#2 [internal function]: React\Http\RequestHeaderParser->feed('GET /data.json ...')

ping @WyriHaximus can this constraint be removed to allow any type of underlying socket?

**Comment by andig** [comment]
Note: for my testing purpose solved by:

        if ($this->remoteSocketUri !== null) {
            $remoteAddress = parse_url($this->remoteSocketUri);

            if (isset($remoteAddress['port'])) {
                $serverParams['REMOTE_ADDR'] = $remoteAddress['host'];
                $serverParams['REMOTE_PORT'] = $remoteAddress['port'];
            }
        }

Host is set to `________` in my tests, so testing for host does not work.

**Comment by clue** [comment]
Thanks for filing this! This is an interesting issue, given that the underlying component does not currently support Unix domain sockets (UDS) (https://github.com/reactphp/socket/issues/25) :-)

I agree that despite this, this is still a minor bug that parsing the request should probably not fail in this case. Do you feel like filing a PR to fix (and test) this? :shipit: 

Considering long-term goals, does it make sense to ignore `REMOTE_ADDR` and `REMOTE_PORT` in this case and/or how do other servers handle this?

**Comment by andig** [comment]
@clue I've added a test. The problem at least on OSX seems to be related to a PHP bug with UDS that I've seen before (https://bugs.php.net/bug.php?id=74556).

**Comment by clue** [comment]
@andig Thanks for the update, I've just commented on this and linked to a related ticket (https://github.com/reactphp/http/pull/234#discussion_r143327946) :+1: 

I wonder where this address you're seeing comes from, as this should already be handled by our socket component (https://github.com/reactphp/socket/pull/100). Do you happen to use an older version or a custom implementation that does not implement the same fix?

**Comment by andig** [comment]
@clue I think the problem is that I'm using the unix socket on the server side, so the `UdsConnector` doesn't even come into play (its what you've pointed out via https://github.com/reactphp/socket/issues/25).

**Comment by clue** [comment]
Thanks for the confirmation! As per reactphp/socket#25 there's currently no official `UnixServer` implementation (PRs welcome!).

If you're using a custom implementation, I would suggest re-using the existing logic to return semantically correct unix address strings. You should be able to re-use the logic from the above link and/or manually return `null` in this case. Can you give this a try and see if this fixes your issue?

As an alternative, once reactphp/socket#25 is resolves (PRs welcome!), we can add an integration test for this and make sure this works out of the box :+1:

**Comment by andig** [comment]
I've opened https://github.com/reactphp/socket/pull/120, currently work in progress. This fixes the problem according to my testing.

Shall we close this PR hence and focus on the proper UDS server? Big thank you for your patience and help!

**Comment by clue** [comment]
Nice! Also happy to hear this appears to fix the issue you're seeing! I think both ways make sense, so I'll leave the decision up to you. Either way, thank you for your continuous effort! :-)

**Comment by andig** [comment]
tl;dr 

Still needs https://github.com/reactphp/socket/pull/123 to get in for Mac-specific failure on current PHP.
